google/cloud/storage/internal/grpc_configure_client_context_test.cc

```diff
@@ -101,6 +102,19 @@ TEST_F(GrpcConfigureClientContext, ApplyQueryParametersQuotaUserAndUserIp) {
   }
 }
 
+TEST_F(GrpcConfigureClientContext, ApplyQueryParametersGrpcOptions) {
+  MockFunction<void(grpc::ClientContext&)> mock;
+  EXPECT_CALL(mock, Call);
+
+  google::cloud::internal::OptionsSpan span(
+      Options{}.set<google::cloud::internal::GrpcSetupOption>(
```

Comment:
A word of caution on what seems to be going on here.

An "internal option" implies to me that we're using thread locals to (conveniently?) plumb state through call layers. 
 I reckon such things would be better passed explicitly.  Options should be reserved for truly optional things from the user.  [Off soap box, and apologies if I'm mistaken.]

Reply:
Ack. The change is confusing on the surface. We are not using `internal::GrpcSetupOption` for state.

Basically, I want to test that current options (supplied by the user) will be used when setting up the `grpc::ClientContext`. But there are no such public options right now. So we test the one option that does anything: `internal::GrpcSetupOption`. In the future, I could see `GrpcSetupOption` being released, or see us adding other options like a `MetadataOption`. https://github.com/googleapis/google-cloud-cpp/issues/4926#issuecomment-1125348091